src/harness/fourslash.ts

```diff
@@ -2910,6 +2910,11 @@ Actual: ${stringify(fullActual)}`);
         }
 
         private verifyDocumentHighlights(expectedRanges: Range[], fileNames: ReadonlyArray<string> = [this.activeFile.fileName]) {
+            expectedRanges = ts.map(expectedRanges, r => {

```

Comment:
It's probably best to treat the expected ranges as readonly, since a test might use the same range object in several different calls: `const [r0, r1] = test.ranges(); test1(r0, r1); test2(r0, r1);` is a pattern I use a lot.

Reply:
Updated to only normalize when reading out of the array, rather than mutating the members of the array.